packages/components/filters/filter-menu/src/trigger-button/trigger-button.tsx

```diff
@@ -0,0 +1,83 @@
+import { type ReactElement } from 'react';
+import IconButton from '@commercetools-uikit/icon-button';
+import { CaretDownIcon, CloseIcon } from '@commercetools-uikit/icons';
+import { Badge } from '../badge';
+import { Chip } from '../chip';
+import type { TAppliedFilterValue } from '../filter-menu';
+
+export type TFilterMenuTriggerButtonProps = {
+  /**
+   * formatted message to display the filter's name
+   */
+  label: string | ReactElement;
+  /**
+   * the values applied to this filter by the user
+   *
+   * NOTE / OPINIONS WELCOME IN PR COMMENTS:
+   * this will almost certainly be some sort of generic w/validation - all the `Chip` really
+   * needs is the selected option's `label`, but these options need to contain a lot more data
+   */
+  appliedFilterValues:
+    | TAppliedFilterValue
+    | TAppliedFilterValue[]
+    | undefined
+    | null;
+  /**
+   * whether or not the filter is disabled
+   */
+  isDisabled?: boolean;
+  /**
+   * indicates whether FilterMenu can be removed from the filtersList
+   */
+  isPersistent?: boolean;
+  /**
+   * controls whether `x` in Trigger Button is displayed - required if `isPersistent` is `false`
+   *
+   * NOTES / OPINIONS WELCOME IN PR COMMENTS:
+   * is the presence/absence of the function sufficient for display logic, or do we need a separate `show-` prop?
+   */
```

Comment:
> is the presence/absence of the function sufficient for display logic, or do we need a separate `show-` prop?

I am not sure it is. It might work but it might explode in our faces as well if things are getting more complex over time.

Alternatively, we could try if it's possible to have those buttons as separate components and use css grid to render them where they are supposed to be in the layout (if they are present in the DOM-structure).

```jsx
<FilterMenu>
  <FilterMenu.Filter>{...}</FilterMenu.Filter>
  <FilterMenu.ClearButton onClick={bleh()}/>
  <FilterMenu.ApplyButton onClick={bleh()}/>
</FilterMenu>
```

Reply:
A couple of random thoughts:
- no matter which direction we go on the presence of a function being sufficient for display logic, we should try to be consistent across components. I personally don't want to set 2 props to get a specific functionality when one will do.
- I've built compound components like Michael's suggesting and I have really liked that pattern. If we are not reusing these components outside of the filter menu, that makes it super easy to understand the code and compartmentalize it.
- I have the same questions about the overlap of "isPersistent" and the presence of the onRemoveFilter function to show the "x" button as I did on the overall filter menu component

Reply:
> I personally don't want to set 2 props to get a specific functionality when one will do.

I just remember that storybook was (and still is) injecting `log` functions in every property that accepts functions but have no value specified in the story-config.

I noticed this when migration a story and it essentially means:  to *not* display one of those buttons in a sb-story we'd have to manually (remember to) set the callback-property to `undefined`.
 
I'm saying this because I could imagine some testing-tools behave similar and this implicit behavior might cause issues further down the line.

> I've built compound components like Michael's suggesting and I have really liked that pattern. If we are not reusing these components outside of the filter menu, that makes it super easy to understand the code and compartmentalize it.

Indeed. We could avoid bloat in the parent components' props and consumers could just render the button if needed and use standard button event-handlers.

Reply:
The approach is fine in abstract, and makes sense in a lot of cases.  I'm not sure this is one of them.

The idea behind having a more restricted `header` and `footer` is to enforce consistent appearance and behavior, as specified in the UX specs.  

Making the buttons available in a compound component relies on consumers to import and declare the buttons on their side, which I think makes consumers more likely to create custom headers and footers because they are declaring the markup.